Sources/MarathonCore/Export.swift

```diff
@@ -0,0 +1,130 @@
+/**
+ *  Marathon
+ *  Copyright (c) John Sundell 2017
+ *  Licensed under the MIT license. See LICENSE file.
+ */
+
+import Foundation
+import Files
+
+// MARK: - Error
+
+public enum ExportError {
+    case missingPath
+    case noMarathonScriptFound
+    case failedToExportScript(String)
+}
+
+extension ExportError: PrintableError {
+    public var message: String {
+        switch self {
+        case .missingPath:
+            return "No script name/path given"
+        case .noMarathonScriptFound:
+            return "No marathon scripts found"
+        case .failedToExportScript(let name):
+            return "Failed to export script \(name)"
+        }
+    }
+    
+    public var hints: [String] {
+        switch self {
+        case .missingPath:
+            return ["Pass the name/path of a script file to export (for example 'marathon export myScript')"]
+        case .noMarathonScriptFound:
+            return ["Marathon can only export scripts managed by marathon"]
+        case .failedToExportScript:
+            return ["Remove directory or provide alternate export path"]
+        }
+    }
+}
+
+// MARK: - Task
+
+internal final class ExportTask: Task, Executable {
+    private typealias Error = ExportError
+    
+    func execute() throws {
+        
+        // Required argument given?
+        guard let scriptNameOrPath = arguments.first?.asScriptPath() else {
+            throw Error.missingPath
+        }
+
+        // Script Name or Full Script Path given?
+        let isScriptName = scriptNameOrPath.components(separatedBy: "/").count == 1
```

Comment:
You comment is on an outdated diff but it is unresolved. Your comment was:

```
Instead of doing this you should be able to just use let script = try scriptManager.script(at: path), like we do in Run, Edit and Install.
```

My reply is:

```
If I do that, then how can I distinguish between a plain swift file and a script managed by marathon or is that not necessary? Should marathon export create a file if it doesn't exist?
```